dash_auth/api_requests.py

```diff
@@ -58,13 +74,81 @@ def _modify_request_kwargs(request_kwargs):
     return copied_kwargs
 
 
+def debug_requests_on(url, **kwargs):
+    # Adapted from https://stackoverflow.com/a/24588289/4142536
+    http_client.HTTPConnection.debuglevel = logging.DEBUG
+    logging.basicConfig()
+    logging.getLogger().setLevel(logging.DEBUG)
+    requests_log = logging.getLogger("requests.packages.urllib3")
+    requests_log.setLevel(logging.DEBUG)
+    requests_log.propagate = True
+    requests_log.addHandler(logging.StreamHandler(sys.stdout))
+
+    # Log some additional info
+    print('Attempting connection at {}'.format(url))
+    parsed_url = urlparse(url)
+    try:
+        print('gettaddrinfo: {}'.format(
+            socket.getaddrinfo(parsed_url.netloc, parsed_url.scheme)
+        ))
+    except Exception as e:
+        print('gettaddrinfo failed')
+        print(e)
+
+
+def debug_requests_off():
+    http_client.HTTPConnection.debuglevel = 0
+    requests_log = logging.getLogger("requests.packages.urllib3")
+    requests_log.setLevel(logging.NOTSET)
+    requests_log.propagate = False
+
+
 def _create_method(method_name):
     def request(path, api_key_auth=True, **request_kwargs):
         copied_kwargs = _modify_request_kwargs(request_kwargs)
-        return getattr(requests, method_name)(
+        request_method = getattr(requests, method_name)
+
+        valid_4xx_status_codes = [404]
+
+        def check_request_before_returning(url, **kwargs):
+            resp = request_method(url, **kwargs)
+
+            # 404's are the only accepted "error" code
+            # as we use this to check if a file exists or not
+            if resp.status_code not in valid_4xx_status_codes:
```

Comment:
🐄 Potential simplification here: `if resp.status_code != 404:`

Since you're only comparing one value, it's clearer and more efficient to just compare that value rather than creating a list and using `not in`.

Not a huge deal though, hence the 🐄 

If you _do_ want to keep `valid_4xx_status_codes`, pep:8ball: says it should be `IN_CAPS` since it's a constant.